tests/neg/i4812.check

```diff
@@ -34,6 +34,12 @@
    |           the type test for A cannot be checked at runtime because it's a local class
    |
    | longer explanation available when compiling with `-explain`
+-- [E092] Pattern Match Error: tests/neg/i4812.scala:80:11 -------------------------------------------------------------
+80 |      case prev: A => prev.elem // error: the type test for A cannot be checked at runtime
+   |           ^
+   |           the type test for A cannot be checked at runtime because it's a local class
```

Comment:
This is a sign you've overreached: you've hit a binding, but it lives inside an unchecked match, so you shouldn't emit the warning.

One way to fix this is doing something like https://github.com/lampepfl/dotty/pull/19190/files, where you record that you're in an unchecked match in the context, so you can honour it before emitting the warning.

Reply:
That's actually simpler than that, we did a mistake when refactoring the code of `interceptTypeApply` by switching the parameters :-(